src/garage/tf/_functions.py

```diff
@@ -0,0 +1,99 @@
+"""Utility functions for tf-based Reinforcement learning algorithms."""
+import numpy as np
+
+from garage.misc import tensor_utils as np_tensor_utils
+from garage.tf.misc import tensor_utils
+
+
+def paths_to_tensors(paths, max_path_length, baseline_predictions, discount,
+                     gae_lambda):
+    """Return processed sample data based on the collected paths.
+
+    Args:
+        paths (list[dict]): A list of collected paths.
+        max_path_length (int): Maximum length of a single rollout.
+        baseline_predictions(numpy.ndarray): : Predicted value of GAE
+            (Generalized Advantage Estimation) Baseline.
+        discount (float): Environment reward discount.
+        gae_lambda (float): Lambda used for generalized advantage
+            estimation.
+
+    Returns:
+        dict: Processed sample data, with key
+            * observations: (numpy.ndarray)
+            * actions: (numpy.ndarray)
+            * rewards: (numpy.ndarray)
+            * baselines: (numpy.ndarray)
+            * returns: (numpy.ndarray)
+            * valids: (numpy.ndarray)
+            * agent_infos: (dict)
+            * env_infos: (dict)
+            * paths: (list[dict])
+
+    """
+    baselines = []
+    returns = []
+    total_steps = 0
+
+    for idx, path in enumerate(paths):
+        total_steps += len(path['rewards'])
+        path_baselines = np.append(baseline_predictions[idx], 0)
+        deltas = (path['rewards'] + discount * path_baselines[1:] -
+                  path_baselines[:-1])
+        path['advantages'] = np_tensor_utils.discount_cumsum(
+            deltas, discount * gae_lambda)
+        path['deltas'] = deltas
```

Comment:
@ryanjulian This helper functions differs from the np version here. So I didn't reuse the np version.

Reply:
alright no problem. I think they'll be unified when we move to TrajectoryBatch soon